console-subscriber/src/visitors.rs

```diff
@@ -119,8 +150,22 @@ impl ResourceVisitor {
     const RES_KIND_FIELD_NAME: &'static str = "kind";
     const RES_KIND_TIMER: &'static str = "timer";
 
-    pub(crate) fn result(self) -> Option<(String, resource::Kind)> {
-        self.concrete_type.zip(self.kind)
+    pub(crate) fn result(self) -> Option<(String, resource::Kind, Option<proto::Location>)> {
+        let concrete_type = self.concrete_type?;
+        let kind = self.kind?;
+
+        let location = if self.file.is_some() && self.line.is_some() && self.column.is_some() {
```

Comment:
i'm on the fence about this, but i think maybe we want to record locations that are "incomplete" (e.g. they are missing some of these fields). we can do this since all the fields on `proto::Location` are optional. I think we probably _don't_ want to record a location if we didn't find a file, since a column/line doesn't make sense without a file, but I think we might want to allow locations that have only file names and no line/column?

what do you think?